### PR TITLE
fix: minor history fixes

### PR DESCRIPTION
**Pull request by rossiam**
* treat integer values for `before` and `after` flags as seconds since the epoch as describe in the help
* refactor: simplified sort function
* refactor: minor reformatting
* refactor: use `confirm` inquirer input type for y/n question instead of text input
* in JSON/YAML output, strictly obey `after`
  * the API first returns items after the specified `after` but allows the user to continue on with the "next" page link
  * the interactive table format output still allows the user to continue on to view items, as the API does
  * when asking for JSON or YAML output, we now strictly obey the `after` flag

## Checklist

- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] Any required documentation has been added
- [x] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: packages/cli/src/lib/commands/history-util.ts [hunk 1]
```diff
@@ -64,36 +62,37 @@ export function getNextDeviceEvents(table: Table, items: DeviceActivity[], optio
...
-				}
+				message: 'Fetch more history records?',
+				default: true,
+			})).more as boolean
+
+			if (!more) {
+				break
+			}
+
+			const table = command.tableGenerator.newOutputTable({ isList: true, head })
```
Comment (Sitlintac) [inline packages/cli/src/lib/commands/history-util.ts:91]: I didn't know you could re-define a `const` in a more restricted context like this.  Neat.  But could certainly get confusing in a less straightforward situation than this one.

File: packages/cli/src/lib/commands/history-util.ts [hunk 2]
```diff
@@ -132,6 +131,11 @@ export const getHistory = async (client: SmartThingsClient, limit: number, perRe
 	const items: DeviceActivity[] = [...history.items]
 	while (items.length < limit && history.hasNext()) {
 		await history.next()
+		// The API allows the user to continue to records before the specified after with paging so
```
Comment (Sitlintac) [inline packages/cli/src/lib/commands/history-util.ts:134]: This wording is a little hard to follow. I think it's "continue to records before the specified after" that confuses me--maybe something like "continue to retrieve records before the specified 'after' date"?

**Review by Sitlintac — APPROVED** [review]

File: packages/cli/src/lib/commands/history-util.ts (hunk 2 above)
Comment (rossiam) [reply packages/cli/src/lib/commands/history-util.ts]: Confusing? That's an understatement. Mangled is more like it. 😄 I must have messed this in editing. I've updated it to:

```
		// The API allows the user to continue to view history from before the specified "after"
		// with paging so we stop processing if we get items that come before the specified "after".
```
